1095/en.md

```diff
@@ -0,0 +1,129 @@
+
+---
+
+
+### Solution
+
+This is a Number Theory (Counting and Derangement) problem. In order to solve this problem you must have some ideas about this topics. 
+
+If you aren't already familiar with How to calculate Binomial Coefficients and Derangement, have a walk through these links:
+
+- [CP-Algorithm: Binomial Coefficients](https://cp-algorithms.com/combinatorics/binomial-coefficients.html)
+- [Brilliant: Derangements](https://brilliant.org/wiki/derangements/)
+
+Let's assume you got a proper knowledge on Binomial Coefficients and Derangements with efficient implementation.
+
+As we have to calculate the number of ways to arrange the numbers so that exactly `K` numbers are in their initial position in first `M` positions, so let's divide this in some parts. 
+
+First let's consider, for the first `M` position, we have `(1..M)` or `M` possible numbers to choose. So we can choose `k` fixed places from `m` possible options in `mCk` ways.
+
+Now as we have K fixed places and they will remain constant in all arrangements, we must make sure there is no other elements in the their initial position among first `M` numbers. So for the remaining positions we have two cases: the position need derangement or we can put any number in that position. 
+
+If we have a close look, we can see there are `M-K` positions, where derangement is must and rest, `M..N` positions doesn't matter what we put.
+
+*Let's observe this part:*
+
+Now let's think how many ways we can put a number in remaining `M-k` positions? we can take any number from `M-K-1` position or remaining `N-M` positions. 
+
+- We take the number from `M-K-1` positions we have two cases:   
+Let's assume `y` is the position in within `M-K-1` from where we are taking number and `x` is the position where we are putting the number. There can be two option:
+
+   - a) We put the number from `y` in place `x`. And you put the number that was in `x` into `y`. So `x` and `y` are already changed now.
+   
+   - b) We put the number from `y` in `X`, but we don't want to put `x`'s number in `y`. That means, it's forbidden for `x`'s number to be in `y`, i.e. `y` still needs to be deranged. 
+-  We take number from `N-M` places so we still have two cases:   
+Let's assume `y` is the position in within `N-M` from where we are taking number and `x` is the position where we are putting the number. There can be two option:
+   - a) We put `y`'s number in `x` and `x`'s number in `y`. So, now you have `M-K-1` numbers to derange and `N-M-1` numbers free.
+   - b)  We put `y`'s number in `x` but don't want to put `x`'s number in `y`. So, we still need to derange `y`. i.e. the number of places to still derange is `M-K-1+1` = `M-K`. But you did use up one of the free numbers from `y`. 
+
+By combining all these derangement observations we can simply come to a final statement:  for each elements from `N-M` choose `x` and derange the rest,`N-K-1` of the elements. 
```

Comment:
here x is just a single element, so we can't subtract the value of x, instead of the position.

Reply:
The observations are the just idea of derangement, anyone with prior knowledge on derangement can simply skip this part.

Reply:
Alright, per your code, I think you are doing the following. Correct me if I am wrong.

There are `n-m` elements where any element `i` can be at index `i` i.e not deranged. For every `x` in `[0, n-m]`, we can choose `x` elements and keep them at their initial indices i.e. not derange them and derange the rest `n-m-x` elements. Since we have `m-k` elements also in need of derangement, we have in total `n-k-x` elements to be deranged. That adds up to

![image](https://user-images.githubusercontent.com/18555205/120304602-be448300-c2f1-11eb-96a4-6de094e97b37.png) 
where `D(n)` is the number of ways to derange `n` elements.

If this is indeed what you're doing, I think your _statement_ at Ln 43 is ambiguous. Please elaborate.

Reply:
Yeah, you are right, the idea is the same. 
I'm updating the line with your ones, is that okay?

Reply:
Yeah it's okay. Let me know.

Reply:
Yeah, I made the mistake in sleepy eyes. Fixed it and thanks for getting it. ^_^